contracts/ERC1238/extensions/ERC1238URIStorage.sol

```diff
@@ -61,64 +69,6 @@ abstract contract ERC1238URIStorage is IERC1238URIStorage, ERC1238 {
         return bytes(_tokenURIs[id]).length > 0;
     }
 
-    /**
-     * @dev Creates `amount` tokens of token type `id` and URI `uri`, and assigns them to `to`.
-     *
-     * Emits a {MintSingle} event.
-     *
-     * Requirements:
-     *
-     * - `to` cannot be the zero address.
-     * - If `to` refers to a smart contract, it must implement {IERC1238Receiver-onERC1238Mint} and return the
-     * acceptance magic value.
-     */
-    function _mintWithURI(
-        address to,
-        uint256 id,
-        uint256 amount,
-        string memory uri,
-        bytes memory data
-    ) internal virtual {
-        _mint(to, id, amount, data);
-        _setTokenURI(id, uri);
-    }
-
-    /**
-     * @dev [Batched] version of {_mintWithURI}.
-     *
-     * Requirements:
-     *
-     * - `ids` and `amounts` must have the same length.
-     * - `ids` and `uris` must have the same length.
-     *
-     * Emits a {MintBatch} event.
-     */
-    function _mintBatchWithURI(
-        address to,
-        uint256[] memory ids,
-        uint256[] memory amounts,
-        string[] memory uris,
-        bytes memory data
-    ) internal virtual {
-        require(to != address(0), "ERC1238: mint to the zero address");
-        require(ids.length == amounts.length, "ERC1238: ids and amounts length mismatch");
-        require(ids.length == uris.length, "ERC1238: ids and URIs length mismatch");
-
-        address minter = msg.sender;
-
-        for (uint256 i = 0; i < ids.length; i++) {
-            _beforeMint(minter, to, ids[i], amounts[i], data);
-
-            _setTokenURI(ids[i], uris[i]);
-
-            _balances[ids[i]][to] += amounts[i];
-        }
-
-        emit MintBatch(minter, to, ids, amounts);
-
-        _doSafeBatchMintAcceptanceCheck(minter, to, ids, amounts, data);
-    }
-
     /**
      * @dev Destroys `amount` of tokens with id `id` owned by `from` and deletes the associated URI.
      *
```

Comment:
Sorry for the weird line that this comment is on. GitHub doesn't let me add comments on certain lines.

Anyway for the function `_burnAndDeleteURI` maybe its enough to just override the base?

```
function _burn(address from, uint256 id, uint256 amount) override internal virtual {
    super._burn(from, id, amount);

   _deleteTokenURI(id);
}
```

Since we don't expect users of the extension to not use this as the main logic for burning.

Reply:
I was thinking actually that in some cases you might want to revoke (burn) a token but leave the URI for third parties to inspect. Imagine it points to a Ceramic Stream that got updated with a revoked status and a reason, you could see that because the URI is still there.
Anyway I'd leave it there for now but I wouldn't mind removing these functions entirely, like I did with the `mintWithUri` ones and leave that as implementation details.

Reply:
Makes sense. Maybe we need to do some thinking on use-case scope before we attack implementations, because I think technical scope shouldn't change based on retrospective potential use-case, since I could see a lot of conversations similar to the ones we had before about rights to burn etc. that I think are born from complexity that arises from conflicting use-cases that clash with our original thoughts/design goals.

Anyways, for now I agree with you on that but we should be very binary in future about what functionalities get folded in where within each context. Here I think we have to frame the thinking around it being the 1238 standard, whereas I think the (very relevant use-case) need of leaving URIs behind after burn is a use-case that is born from our own specific needs. In these cases I think we should move away from scope creep and not bundle everything into one place. Our own specific need should be solved separately, either as a further extension or some other method (or even other standard) so as to not infect the more abstract and cleaner design of something that should be use-case agnostic and fairly lightweight and unopinionated.

Sorry for rant :D

Reply:
After a bit more thinking I think your idea might be the most elegant. Since you're right that the logic behind `burnAndDeleteURI` is more an implementation detail that can be added by the implementor and does not need to be part of the standard for us. In this way the `Badge.sol` burn function simply does the same boolean check and just calls `_deleteTokenURI` and then continues the usual `burn` flow.

Reply:
So keeping symmetry I think is key here.

If the lifecycle of the token itself doesn't deal with the URI then no part of the lifecycle management (mint/burn) should touch it.

This leaves role symmetry and lifecycle management symmetry where mint/burn controls token existence and set/forget controls tokenURI details.

In this way I think we either put both `burnAndDeleteURI` and `mintWithURI` as functions in the example `Badge.sol` or neither.

Reply:
Agree, let's stay consistent, so I moved the burn + delete URI functions to the Badge example